### DISCUSSION TITLE
Peridynamic Heat Conduction Models

### DISCUSSION THREAD

**Discussion by rbh007720** [idaholab/moose#20981] [Q&A Modules: General] [answered]
Is the Bond-based model able to solve the transient heat transfer problem? In "Thermomechanical peridynamic analysis with irregular non-uniform domain discretization", the implementation is based on an implicit solver for the steady-state problem. Is it possible to treat unsteady-state heat conduction?

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
@laagesen @bwspenc any ideas on this ?

**Reply by jiangwen84** [reply]
@rbh007720 have you looked at any PD time derivative kernel? 

Let me @hchen139 .

**Reply by hchen139** [reply]
@rbh007720 I don't think we have implemented any time derivative kernel for PD yet. You can look at the TimeDerivativeNodalKernel to consider the transient heat transfer process.

**Reply by rbh007720** [reply]
Thanks for the reply. Should I look at TimeDerivativeNodalKernel or TimeDerivative Kernel? I would like to know what is the difference between them.
Thanks!

**Reply by hchen139** [reply]
TimeDerivativeNodalKernel operates on nodes while TimeDerivative operates on elements. Since PD is a Lagrangian method and uses nodal integration, nodal kernel would make sense for the transient kernel. All the current implementations of PD kernels are based on elemental kernel to use the material classes in other MOOSE modules that are based on FEM.

**Reply by GiudGiud** [reply]
Nodal kernels are made to solve ODEs that live on individual nodes. So it's quite different from FEM. If you only had nodal kernels, you would only be solving decoupled ODEs on each node. Of course they may be coupled through the values of FEM fields (that have regular kernels for example). 
So probably TimeDerivative

**Comment by Narayanan1999** [comment]
[rbh007720](https://github.com/rbh007720) Sir, were you able to do transient analysis in Peridynamics using TimeDerivativeNodalKernel? If so, can you please share the input file or that transient part code?
I tried solving a 1D bar subjected to an initial strain. The transient analysis is done using Newmark-Beta Method from Tensor Mechanics module. Though the input file runs, the values are not matching with theoretical value. The following is the result that I obtained for the displacement occurred at the midpoint of the bar during this time interval (x-time; y-displacment):
![image](https://github.com/idaholab/moose/assets/83218826/5f29fb99-c55d-4108-8887-e510e03eda19)

**Reply by hchen139** [reply]
I am not sure whether you can directly run transient analysis based on the available developments in the peridynamics module. There might need some additional developments.

**Reply by Narayanan1999** [reply]
Ok Sir, Thank you.
